streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -125,6 +119,7 @@ public void testMetrics() {
         final Map<String, String> metricTags = new LinkedHashMap<>();
         metricTags.put("processor-node-id", node.name());
         metricTags.put("task-id", context.taskId().toString());
+        metricTags.put("client-id", "mock");
```

Comment:
This is required per the docs, but we previously only added it in production code paths. Now we add it in all code paths.